File: src/main/java/org/oewntk/ser/in/CoreFactory.java
/*
 * Copyright (c) 2021-2021. Bernard Bou.
 */

package org.oewntk.ser.in;

import org.oewntk.model.CoreModel;
import org.oewntk.model.DeSerialize;

import java.io.File;
import java.io.IOException;
import java.util.function.Supplier;

public class CoreFactory implements Supplier<CoreModel>
{
	private final File file;

	public CoreFactory(final File file)
	{
		this.file = file;
	}

	@Override
	public CoreModel get()
	{
		try
		{
			return DeSerialize.deSerializeCoreModel(file);
		}
		catch (IOException | ClassNotFoundException e)
		{
			e.printStackTrace(Tracing.psErr);
			return null;
		}
	}

	static public CoreModel makeCoreModel(String[] args)
	{
		File file = new File(args[0]);
		return new CoreFactory(file).get();
	}

	static public void main(String[] args)
	{
		CoreModel model = makeCoreModel(args);
		Tracing.psInfo.printf("[CoreModel] %s\n%s\n%s%n", model.getSource(), model.info(), model.counts());
	}
}

File: src/main/java/org/oewntk/ser/in/CoreFactory.kt
/*
 * Copyright (c) 2021-2024. Bernard Bou.
 */
package org.oewntk.ser.`in`

import org.oewntk.model.CoreModel
import org.oewntk.model.DeSerialize.deSerializeCoreModel
import org.oewntk.model.ModelInfo
import java.io.File
import java.io.IOException
import java.util.function.Supplier

/**
 * Core model factory from serialization
 *
 * @property file serialization file
 */
class CoreFactory(
    private val file: File,
) : Supplier<CoreModel?> {

    override fun get(): CoreModel? {
        try {
            return deSerializeCoreModel(file)
        } catch (e: IOException) {
            e.printStackTrace(Tracing.psErr)
            return null
        } catch (e: ClassNotFoundException) {
            e.printStackTrace(Tracing.psErr)
            return null
        }
    }

    companion object {

        /**
         * Make core model from serialization
         *
         * @param args command-line arguments
         * @return core model
         */
        private fun makeCoreModel(args: Array<String>): CoreModel? {
            val file = File(args[0])
            return CoreFactory(file).get()
        }

        /**
         * Main
         *
         * @param args command-line arguments
         */
        @JvmStatic
        fun main(args: Array<String>) {
            val model = makeCoreModel(args)
            Tracing.psInfo.printf("[CoreModel] %s%n%s%n%s%n", model!!.source, model.info(), ModelInfo.counts(model))
        }
    }
}